applications/tari_base_node/src/builder.rs

```diff
@@ -495,43 +493,155 @@ where
             };
 
             if bootstrapped {
+                debug!(
+                    target: LOG_TARGET,
+                    "Initial sync achieved and starting with transaction and UTXO validation protocols.",
+                );
                 let _ = transaction_service_handle
                     .restart_broadcast_protocols()
                     .await
                     .map_err(|e| {
                         error!(
                             target: LOG_TARGET,
-                            "Problem restarting broadcast protocols in the Transaction Service"
+                            "Problem restarting broadcast protocols in the Transaction Service: {}", e
                         );
                         e
                     });
-
                 let _ = transaction_service_handle
                     .restart_transaction_protocols()
                     .await
                     .map_err(|e| {
                         error!(
                             target: LOG_TARGET,
-                            "Problem restarting transaction negotiation protocols in the Transaction Service"
+                            "Problem restarting transaction negotiation protocols in the Transaction Service: {}", e
                         );
                         e
                     });
+
+                #[derive(PartialEq, Debug)]
+                enum ValidationTask {
+                    InvalidOutputs,
+                    UnspentOutputs,
+                }
+                let mut event_stream = oms_handle.get_event_stream_fused();
+                let mut validation = ValidationTask::InvalidOutputs;
+                let mut end_protocol = false;
+                let mut delay = delay_for(config_base_node_query_timeout_clone * 2).fuse();
+                loop {
+                    match validation {
+                        ValidationTask::InvalidOutputs => {
+                            // Setting the base node public key starts the protocol
+                            let _ = oms_handle
+                                .clone()
+                                .set_base_node_public_key(base_node_public_key.clone())
+                                .await
+                                .map_err(|e| {
+                                    error!(
+                                        target: LOG_TARGET,
+                                        "Problem with Output Manager Service setting the base node public key: {}", e
+                                    );
+                                    e
+                                });
+                            trace!(target: LOG_TARGET, "Attempting UTXO validation for Invalid Outputs.",);
+                        },
+                        ValidationTask::UnspentOutputs => {
+                            let _ = oms_handle
```

Comment:
So you are throwing away the ID of the validation request here. The number returned by this call is what you should use to differentiate between the OutputManagerEvents while you wait this protocol to resolve. This way you don't have to be careful to only start one protocol at a time because you can tell which events are the ones you care about using this ID.

Reply:
See comment above.